slatedb/src/manifest/store.rs

```diff
@@ -682,6 +682,16 @@ impl ManifestStore {
         self.try_read_manifest(id).await?.ok_or(ManifestMissing(id))
     }
 
+    pub(crate) async fn validate_no_wal_object_store_configured(&self) -> Result<(), SlateDBError> {
```

Comment:
Nit: Add docs explaining what this is and why we need it.

Reply:
https://github.com/slatedb/slatedb/pull/558/commits/e6b327729d73f2e34eff970c3213a639e33f915d